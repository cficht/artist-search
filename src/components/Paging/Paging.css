.Paging {
  clear: left;
  display: flex;
  justify-content: center;  
  align-items: center;
  margin: 10px 0px 10px 0px;
  padding: 10px;
  border-radius: 10px;
  background-color: #ffb367;

  button {
    margin: 5px 15px 5px 15px;
    background-color: #ffe277;
    color: black;
    font-size: 16px;
    text-align: center;
  }

  button:disabled {
    background-color: #ffe2bc;
  }
}