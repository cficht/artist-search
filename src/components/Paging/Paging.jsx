import React from 'react';
import PropTypes from 'prop-types';

const Paging = ({ page, list, handlePage }) => (
  <footer>
    <button onClick={() => handlePage(-1)} disabled={page === 0}>Back</button>
    <button onClick={() => handlePage(1)} disabled={list.length < 25}>Forward</button>
  </footer>
);

Paging.propTypes = {
  page: PropTypes.number.isRequired,
  list: PropTypes.array.isRequired,
  handlePage: PropTypes.func.isRequired
};

export default Paging;
