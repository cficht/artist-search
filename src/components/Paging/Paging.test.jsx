import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging component', () => {
  it('renders Paging', () => {
    const wrapper = shallow(<Paging 
      page={0}
      list={[]}
      handlePage={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
