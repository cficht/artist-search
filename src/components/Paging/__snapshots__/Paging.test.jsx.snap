// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<footer>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Back
  </button>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Forward
  </button>
</footer>
`;
